packages/go_router/lib/src/match.dart

```diff
@@ -136,16 +138,16 @@ class ImperativeRouteMatch extends RouteMatch {
     completer.complete(value);
   }
 
-  // An ImperativeRouteMatch has its own life cycle due the the _completer.
-  // comparing _completer between instances would be the same thing as
-  // comparing object reference.
   @override
   bool operator ==(Object other) {
```

Comment:
Previous there are no use case that the imperative route match can be equal because all the completers in them are different. so I optimize it away to use identity hash (otherwise it would use the super == which will return incorrect result).

Now that the RouteMatchList will be reparsed after the routing config changes. The ImperativeRouteMatch will reuse its completer when them are recreated, so there can be cases where equality is possible.